go/staking/tests/tester.go

```diff
@@ -83,45 +155,40 @@ func StakingClientImplementationTests(t *testing.T, backend api.Backend, consens
 	} {
 		t.Run(tc.n, func(t *testing.T) { tc.fn(t, backend, consensus) })
 	}
+
+	// Update tests' state for subsequent node tests (which re-use the state).
+	state.update(backend, consensus)
 }
 
 func testInitialEnv(t *testing.T, backend api.Backend, consensus consensusAPI.Backend) {
 	require := require.New(t)
 
 	totalSupply, err := backend.TotalSupply(context.Background(), consensusAPI.HeightLatest)
 	require.NoError(err, "TotalSupply")
-	require.Equal(&testTotalSupply, totalSupply, "TotalSupply - value")
+	require.Equal(state.totalSupply, totalSupply, "TotalSupply - value")
 
 	accounts, err := backend.Accounts(context.Background(), consensusAPI.HeightLatest)
 	require.NoError(err, "Accounts")
-	// Since all operations now cause nonce updates, we have some more accounts here:
-	//
-	//   1 is from the genesis file
-	//   1 is the mock epochtime test signer
-	//   1 is the test node worker identity
-	//   1 is the test node entity signer
-	//   1 is the test entity signer
-	//
-	require.Len(accounts, 5, "Accounts - nr entries")
-	require.Contains(accounts, SrcID, "Accounts[0] == testID")
+	require.Equal(state.nAccounts, len(accounts), "Accounts - number")
+	require.Contains(accounts, SrcID, "Accounts - testID one of them")
 
 	acc, err := backend.AccountInfo(context.Background(), &api.OwnerQuery{Owner: SrcID, Height: consensusAPI.HeightLatest})
 	require.NoError(err, "src: AccountInfo")
-	require.Equal(testTotalSupply, acc.General.Balance, "src: general balance")
-	require.True(acc.Escrow.Active.Balance.IsZero(), "src: active escrow balance")
+	require.Equal(state.srcAccountGeneralBalance, acc.General.Balance, "src: general balance")
+	require.Equal(state.srcAccountEscrowActiveBalance, acc.Escrow.Active.Balance, "src: active escrow balance")
 	require.True(acc.Escrow.Debonding.Balance.IsZero(), "src: debonding escrow balance")
-	require.EqualValues(0, acc.General.Nonce, "src: nonce")
+	require.EqualValues(state.srcAccountNonce, acc.General.Nonce, "src: nonce")
```

Comment:
:slightly_smiling_face:  The tests' state is initialized via fixed values (i.e. without calling `update()`).

The subsequent runs are indeed tautological since the `update()` method uses the same functions to update the state as are used here to check the state.

Should we only run `testInitialEnv()` on the first run of the tests? Should we skip it altogether?